### PR TITLE
MULE-16208 - Find resources is not taking into account lookup of file…

### PR DESCRIPTION
**Pull request by pablolagreca**
…s by directory

### CODE REVIEW

**Comment by pablolagreca** [comment]
--validate

File: modules/artifact/src/main/java/org/mule/runtime/module/artifact/api/classloader/RegionClassLoader.java [hunk 1]
```diff
@@ -339,19 +340,30 @@ public final URL findResource(final String name) {
   @Override
   public final Enumeration<URL> findResources(final String name) throws IOException {
```
Comment (sonar-mulesoft) [inline modules/artifact/src/main/java/org/mule/runtime/module/artifact/api/classloader/RegionClassLoader.java]: Code Smell Code Smell: Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. ([squid:S3776](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS3776))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7561&issues=AWhqiKMiotIXnmsUkVkL&open=AWhqiKMiotIXnmsUkVkL)

[ISSUE_KEY]: AWhqiKMiotIXnmsUkVkL

Comment (pablolagreca) [reply modules/artifact/src/main/java/org/mule/runtime/module/artifact/api/classloader/RegionClassLoader.java]: --validate

**Comment by pablolagreca** [comment]
--validate

File: modules/artifact/src/main/java/org/mule/runtime/module/artifact/api/classloader/RegionClassLoader.java (hunk 1 above)
Comment (sonar-mulesoft) [inline modules/artifact/src/main/java/org/mule/runtime/module/artifact/api/classloader/RegionClassLoader.java]: Code Smell Code Smell: Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. ([squid:S3776](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS3776))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7561&issues=AWhqiKMiotIXnmsUkVkL&open=AWhqiKMiotIXnmsUkVkL)

[ISSUE_KEY]: AWhqiKMiotIXnmsUkVkL

**Pull request merged**